src/pr-trigger.ts

```diff
@@ -98,22 +97,7 @@ export async function httpTrigger(context: Context, req: HttpRequest) {
         if (!numberInURL) throw new Error(`Could not get PR for project card URL: ${cardURL}`);
         prNumber = +numberInURL[0].substring(1);
     } else if ("check_suite" in webhook) {
-        // See https://github.com/maintainers/early-access-feedback/issues/114 for more context on getting a PR from a SHA
-        // TLDR: it's not in the API, and this search hack has been in used on Peril for the last ~3 years
-        // => Instead of an arbitrary search, use `associatedPullRequests` (https://developer.github.com/v4/changelog/2019-03-08-schema-changes/)
-        const owner = webhook.repository.owner.login;
-        const repo = webhook.repository.name;
-        const sha = webhook.check_suite.head_sha;
-        const pr = await runQueryToGetPRMetadataForSHA1(owner, repo, sha);
-        if (!pr || pr.closed) {
-            const whatFailed = !pr ? "a PR" : "an open PR";
-            context.log.info(`Skipped webhook, could not find ${whatFailed} for the sha referenced in the status (${sha})`);
-            context.res = {
-                status: 204,
-                body: `NOOPing due to not finding ${whatFailed} for the sha ${sha}`
-            };
-            return;
-        }
+        const [pr] = webhook.check_suite.pull_requests;
```

Comment:
1. I prefer the shorter and more readable (IMO): `prNumber = webhook.check_suite.pull_requests[0]`
2. Is it guaranteed to work properly?  (I vaguely remember a lot of issues getting this to work)

Reply:
> 1. I prefer the shorter and more readable (IMO): `prNumber = webhook.check_suite.pull_requests[0]`

Done :heavy_check_mark:

> 2. Is it guaranteed to work properly?  (I vaguely remember a lot of issues getting this to work)

I did see the comments mentioning issues with the `associatedPullRequests` GraphQL field, however nothing to indicate trouble with the `check_suite.pull_requests` payload property?
https://github.com/DefinitelyTyped/dt-mergebot/blob/ea44dfba0e070fd1c79e259f977f21ab34e72ea1/src/pr-trigger.ts#L101-L103
https://github.com/DefinitelyTyped/dt-mergebot/blob/ea44dfba0e070fd1c79e259f977f21ab34e72ea1/src/queries/SHA1-to-PR-query.ts#L28

Reply:
Just that I remember trying to look through stuff and see if the PR number is in a place that can be used.  But maybe I missed this -- so hopefully it'll work fine.